### ISSUE TITLE
crash on /list

### ISSUE THREAD

**Issue by DarthGandalf** [kvirc/KVIrc#1093]
**Reported by ambossarm on 09 Feb 2011 21:23:05 UTC**
I was informed that on ubuntu lucid r5381 crashes on /list, I tried it with r5438 and it still crashes.

My ubuntu karmic machine also crashes. Tried it on freenode and quakenet

```

 gdb kvirc
GNU gdb (GDB) 7.1-ubuntu
Copyright (C) 2010 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>...
Reading symbols from /usr/bin/kvirc...Reading symbols from /usr/lib/debug/usr/bin/kvirc...done.
done.
(gdb) run
Starting program: /usr/bin/kvirc 
[Thread debugging using libthread_db enabled]
[New Thread 0x7fffde30c700 (LWP 29284)]
[Thread 0x7fffde30c700 (LWP 29284) exited]


Program received signal SIGSEGV, Segmentation fault.
0x00007ffff4ea19a0 in QString::indexOf(QChar, int, Qt::CaseSensitivity) const
    () from /usr/lib/libQtCore.so.4



(gdb) bt
#0  0x00007ffff4ea19a0 in QString::indexOf(QChar, int, Qt::CaseSensitivity) const () from /usr/lib/libQtCore.so.4
#1  0x00007ffff5e8ed57 in QFontMetrics::width(QString const&, int) const ()
   from /usr/lib/libQtGui.so.4
#2  0x00007fffdd4e88be in ChannelTreeWidgetItemDelegate::sizeHint(QStyleOptionViewItem const&, QModelIndex const&) const ()
   from /usr/lib/kvirc/4.1/modules/libkvilist.so
#3  0x00007ffff61bab28 in QTreeView::indexRowSizeHint(QModelIndex const&) const
    () from /usr/lib/libQtGui.so.4
#4  0x00007ffff61c3da7 in QTreeView::rowsInserted(QModelIndex const&, int, int)
    () from /usr/lib/libQtGui.so.4
#5  0x00007ffff61804ad in QAbstractItemView::qt_metacall(QMetaObject::Call, int, void**) () from /usr/lib/libQtGui.so.4
#6  0x00007ffff61c9775 in QTreeView::qt_metacall(QMetaObject::Call, int, void**) () from /usr/lib/libQtGui.so.4
#7  0x00007ffff6200de5 in QTreeWidget::qt_metacall(QMetaObject::Call, int, void**) () from /usr/lib/libQtGui.so.4
#8  0x00000000006696e0 in KviThemedTreeWidget::qt_metacall(QMetaObject::Call, int, void**) ()
#9  0x00007ffff4f66e3f in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () from /usr/lib/libQtCore.so.4
#10 0x00007ffff4fb2314 in QAbstractItemModel::rowsInserted(QModelIndex const&, int, int) () from /usr/lib/libQtCore.so.4
---Type <return> to continue, or q <return> to quit---
#11 0x00007ffff4f4d087 in QAbstractItemModel::endInsertRows() ()
   from /usr/lib/libQtCore.so.4
#12 0x00007ffff61fb44c in QTreeWidgetItem::insertChild(int, QTreeWidgetItem*)
    () from /usr/lib/libQtGui.so.4
#13 0x00007ffff61fb5fb in QTreeWidgetItem::addChild(QTreeWidgetItem*) ()
   from /usr/lib/libQtGui.so.4
#14 0x00007ffff61fbf1d in QTreeWidgetItem::QTreeWidgetItem(QTreeWidget*, int)
    () from /usr/lib/libQtGui.so.4
#15 0x00007fffdd4e8597 in ChannelTreeWidgetItem::ChannelTreeWidgetItem(QTreeWidget*, ChannelTreeWidgetItemData*) ()
   from /usr/lib/kvirc/4.1/modules/libkvilist.so
#16 0x00007fffdd4e8c71 in ListWindow::flush() ()
   from /usr/lib/kvirc/4.1/modules/libkvilist.so
#17 0x00007fffdd4ec094 in ListWindow::qt_metacall(QMetaObject::Call, int, void**) () from /usr/lib/kvirc/4.1/modules/libkvilist.so
#18 0x00007ffff4f66e3f in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () from /usr/lib/libQtCore.so.4
#19 0x00007ffff4f63a63 in QObject::event(QEvent*) ()
   from /usr/lib/libQtCore.so.4
#20 0x00007ffff5c5922c in QApplicationPrivate::notify_helper(QObject*, QEvent*)
    () from /usr/lib/libQtGui.so.4
#21 0x00007ffff5c5f6fb in QApplication::notify(QObject*, QEvent*) ()
   from /usr/lib/libQtGui.so.4
---Type <return> to continue, or q <return> to quit---
#22 0x00007ffff7264a16 in KApplication::notify(QObject*, QEvent*) ()
   from /usr/lib/libkdeui.so.5
#23 0x00007ffff4f5406c in QCoreApplication::notifyInternal(QObject*, QEvent*)
    () from /usr/lib/libQtCore.so.4
#24 0x00007ffff4f80d42 in ?? () from /usr/lib/libQtCore.so.4
#25 0x00007ffff4f7d848 in ?? () from /usr/lib/libQtCore.so.4
#26 0x00007ffff0f518c2 in g_main_context_dispatch () from /lib/libglib-2.0.so.0
#27 0x00007ffff0f55748 in ?? () from /lib/libglib-2.0.so.0
#28 0x00007ffff0f558fc in g_main_context_iteration ()
   from /lib/libglib-2.0.so.0
#29 0x00007ffff4f7d513 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQtCore.so.4
#30 0x00007ffff5d0946e in ?? () from /usr/lib/libQtGui.so.4
#31 0x00007ffff4f52992 in QEventLoop::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQtCore.so.4
#32 0x00007ffff4f52d6c in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQtCore.so.4
#33 0x00007ffff4f56aab in QCoreApplication::exec() ()
   from /usr/lib/libQtCore.so.4
#34 0x0000000000643253 in main ()



(gdb) bt full
#0  0x00007ffff4ea19a0 in QString::indexOf(QChar, int, Qt::CaseSensitivity) const () from /usr/lib/libQtCore.so.4
No symbol table info available.
#1  0x00007ffff5e8ed57 in QFontMetrics::width(QString const&, int) const ()
   from /usr/lib/libQtGui.so.4
No symbol table info available.
#2  0x00007fffdd4e88be in ChannelTreeWidgetItemDelegate::sizeHint(QStyleOptionViewItem const&, QModelIndex const&) const ()
   from /usr/lib/kvirc/4.1/modules/libkvilist.so
No symbol table info available.
#3  0x00007ffff61bab28 in QTreeView::indexRowSizeHint(QModelIndex const&) const
    () from /usr/lib/libQtGui.so.4
No symbol table info available.
#4  0x00007ffff61c3da7 in QTreeView::rowsInserted(QModelIndex const&, int, int)
    () from /usr/lib/libQtGui.so.4
No symbol table info available.
#5  0x00007ffff61804ad in QAbstractItemView::qt_metacall(QMetaObject::Call, int, void**) () from /usr/lib/libQtGui.so.4
No symbol table info available.
#6  0x00007ffff61c9775 in QTreeView::qt_metacall(QMetaObject::Call, int, void**) () from /usr/lib/libQtGui.so.4
No symbol table info available.
#7  0x00007ffff6200de5 in QTreeWidget::qt_metacall(QMetaObject::Call, int, void*---Type <return> to continue, or q <return> to quit---
*) () from /usr/lib/libQtGui.so.4
No symbol table info available.
#8  0x00000000006696e0 in KviThemedTreeWidget::qt_metacall(QMetaObject::Call, int, void**) ()
No symbol table info available.
#9  0x00007ffff4f66e3f in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () from /usr/lib/libQtCore.so.4
No symbol table info available.
#10 0x00007ffff4fb2314 in QAbstractItemModel::rowsInserted(QModelIndex const&, int, int) () from /usr/lib/libQtCore.so.4
No symbol table info available.
#11 0x00007ffff4f4d087 in QAbstractItemModel::endInsertRows() ()
   from /usr/lib/libQtCore.so.4
No symbol table info available.
#12 0x00007ffff61fb44c in QTreeWidgetItem::insertChild(int, QTreeWidgetItem*)
    () from /usr/lib/libQtGui.so.4
No symbol table info available.
#13 0x00007ffff61fb5fb in QTreeWidgetItem::addChild(QTreeWidgetItem*) ()
   from /usr/lib/libQtGui.so.4
No symbol table info available.
#14 0x00007ffff61fbf1d in QTreeWidgetItem::QTreeWidgetItem(QTreeWidget*, int)
    () from /usr/lib/libQtGui.so.4
No symbol table info available.
---Type <return> to continue, or q <return> to quit---
#15 0x00007fffdd4e8597 in ChannelTreeWidgetItem::ChannelTreeWidgetItem(QTreeWidget*, ChannelTreeWidgetItemData*) ()
   from /usr/lib/kvirc/4.1/modules/libkvilist.so
No symbol table info available.
#16 0x00007fffdd4e8c71 in ListWindow::flush() ()
   from /usr/lib/kvirc/4.1/modules/libkvilist.so
No symbol table info available.
#17 0x00007fffdd4ec094 in ListWindow::qt_metacall(QMetaObject::Call, int, void**) () from /usr/lib/kvirc/4.1/modules/libkvilist.so
No symbol table info available.
#18 0x00007ffff4f66e3f in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () from /usr/lib/libQtCore.so.4
No symbol table info available.
#19 0x00007ffff4f63a63 in QObject::event(QEvent*) ()
   from /usr/lib/libQtCore.so.4
No symbol table info available.
#20 0x00007ffff5c5922c in QApplicationPrivate::notify_helper(QObject*, QEvent*)
    () from /usr/lib/libQtGui.so.4
No symbol table info available.
#21 0x00007ffff5c5f6fb in QApplication::notify(QObject*, QEvent*) ()
   from /usr/lib/libQtGui.so.4
No symbol table info available.
#22 0x00007ffff7264a16 in KApplication::notify(QObject*, QEvent*) ()
---Type <return> to continue, or q <return> to quit---
   from /usr/lib/libkdeui.so.5
No symbol table info available.
#23 0x00007ffff4f5406c in QCoreApplication::notifyInternal(QObject*, QEvent*)
    () from /usr/lib/libQtCore.so.4
No symbol table info available.
#24 0x00007ffff4f80d42 in ?? () from /usr/lib/libQtCore.so.4
No symbol table info available.
#25 0x00007ffff4f7d848 in ?? () from /usr/lib/libQtCore.so.4
No symbol table info available.
#26 0x00007ffff0f518c2 in g_main_context_dispatch () from /lib/libglib-2.0.so.0
No symbol table info available.
#27 0x00007ffff0f55748 in ?? () from /lib/libglib-2.0.so.0
No symbol table info available.
#28 0x00007ffff0f558fc in g_main_context_iteration ()
   from /lib/libglib-2.0.so.0
No symbol table info available.
#29 0x00007ffff4f7d513 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQtCore.so.4
No symbol table info available.
#30 0x00007ffff5d0946e in ?? () from /usr/lib/libQtGui.so.4
No symbol table info available.
#31 0x00007ffff4f52992 in QEventLoop::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQtCore.so.4
---Type <return> to continue, or q <return> to quit---
No symbol table info available.
#32 0x00007ffff4f52d6c in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQtCore.so.4
No symbol table info available.
#33 0x00007ffff4f56aab in QCoreApplication::exec() ()
   from /usr/lib/libQtCore.so.4
No symbol table info available.
#34 0x0000000000643253 in main ()
No symbol table info available.
(gdb) 


```

---

Migrated from: https://svn.kvirc.de/kvirc/ticket/1093

**Comment by DarthGandalf** [comment]
**Comment by pragmaware on 09 Feb 2011 22:43:04**
Might be fixed in trunk. Can you test ?

---

Resolution: none → fixed
Status: new → closed

**Comment by DarthGandalf** [comment]
**Comment by ambossarm on 09 Feb 2011 23:22:46**
r5443 still crashes

```
(gdb) bt
#0  0x00007ffff4ea19a0 in QString::indexOf(QChar, int, Qt::CaseSensitivity) const () from /usr/lib/libQtCore.so.4
#1  0x00007ffff5e8ed57 in QFontMetrics::width(QString const&, int) const ()
   from /usr/lib/libQtGui.so.4
#2  0x00007fffdd4eaa57 in ChannelTreeWidgetItemDelegate::sizeHint(QStyleOptionViewItem const&, QModelIndex const&) const ()
   from /usr/lib/kvirc/4.1/modules/libkvilist.so
#3  0x00007ffff61bab28 in QTreeView::indexRowSizeHint(QModelIndex const&) const
    () from /usr/lib/libQtGui.so.4
#4  0x00007ffff61c3da7 in QTreeView::rowsInserted(QModelIndex const&, int, int)
    () from /usr/lib/libQtGui.so.4
#5  0x00007ffff61804ad in QAbstractItemView::qt_metacall(QMetaObject::Call, int, void**) () from /usr/lib/libQtGui.so.4
#6  0x00007ffff61c9775 in QTreeView::qt_metacall(QMetaObject::Call, int, void**) () from /usr/lib/libQtGui.so.4
#7  0x00007ffff6200de5 in QTreeWidget::qt_metacall(QMetaObject::Call, int, void**) () from /usr/lib/libQtGui.so.4
#8  0x00000000006696e0 in KviThemedTreeWidget::qt_metacall(QMetaObject::Call, int, void**) ()
#9  0x00007ffff4f66e3f in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () from /usr/lib/libQtCore.so.4
#10 0x00007ffff4fb2314 in QAbstractItemModel::rowsInserted(QModelIndex const&, int, int) () from /usr/lib/libQtCore.so.4
---Type <return> to continue, or q <return> to quit---
#11 0x00007ffff4f4d087 in QAbstractItemModel::endInsertRows() ()
   from /usr/lib/libQtCore.so.4
#12 0x00007ffff61fb44c in QTreeWidgetItem::insertChild(int, QTreeWidgetItem*)
    () from /usr/lib/libQtGui.so.4
#13 0x00007ffff61fb5fb in QTreeWidgetItem::addChild(QTreeWidgetItem*) ()
   from /usr/lib/libQtGui.so.4
#14 0x00007ffff61fbf1d in QTreeWidgetItem::QTreeWidgetItem(QTreeWidget*, int)
    () from /usr/lib/libQtGui.so.4
#15 0x00007fffdd4e8597 in ChannelTreeWidgetItem::ChannelTreeWidgetItem(QTreeWidget*, ChannelTreeWidgetItemData*) ()
   from /usr/lib/kvirc/4.1/modules/libkvilist.so
#16 0x00007fffdd4e8b31 in ListWindow::flush() ()
   from /usr/lib/kvirc/4.1/modules/libkvilist.so
#17 0x00007fffdd4ec0b4 in ListWindow::qt_metacall(QMetaObject::Call, int, void**) () from /usr/lib/kvirc/4.1/modules/libkvilist.so
#18 0x00007ffff4f66e3f in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () from /usr/lib/libQtCore.so.4
#19 0x00007ffff4f63a63 in QObject::event(QEvent*) ()
   from /usr/lib/libQtCore.so.4
#20 0x00007ffff5c5922c in QApplicationPrivate::notify_helper(QObject*, QEvent*)
    () from /usr/lib/libQtGui.so.4
#21 0x00007ffff5c5f6fb in QApplication::notify(QObject*, QEvent*) ()
   from /usr/lib/libQtGui.so.4
---Type <return> to continue, or q <return> to quit---
#22 0x00007ffff7264a16 in KApplication::notify(QObject*, QEvent*) ()
   from /usr/lib/libkdeui.so.5
#23 0x00007ffff4f5406c in QCoreApplication::notifyInternal(QObject*, QEvent*)
    () from /usr/lib/libQtCore.so.4
#24 0x00007ffff4f80d42 in ?? () from /usr/lib/libQtCore.so.4
#25 0x00007ffff4f7d848 in ?? () from /usr/lib/libQtCore.so.4
#26 0x00007ffff0f518c2 in g_main_context_dispatch () from /lib/libglib-2.0.so.0
#27 0x00007ffff0f55748 in ?? () from /lib/libglib-2.0.so.0
#28 0x00007ffff0f558fc in g_main_context_iteration ()
   from /lib/libglib-2.0.so.0
#29 0x00007ffff4f7d513 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQtCore.so.4
#30 0x00007ffff5d0946e in ?? () from /usr/lib/libQtGui.so.4
#31 0x00007ffff4f52992 in QEventLoop::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQtCore.so.4
#32 0x00007ffff4f52d6c in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQtCore.so.4
#33 0x00007ffff4f56aab in QCoreApplication::exec() ()
   from /usr/lib/libQtCore.so.4
#34 0x0000000000643253 in main ()



(gdb) bt full
#0  0x00007ffff4ea19a0 in QString::indexOf(QChar, int, Qt::CaseSensitivity) const () from /usr/lib/libQtCore.so.4
No symbol table info available.
#1  0x00007ffff5e8ed57 in QFontMetrics::width(QString const&, int) const ()
   from /usr/lib/libQtGui.so.4
No symbol table info available.
#2  0x00007fffdd4eaa57 in ChannelTreeWidgetItemDelegate::sizeHint(QStyleOptionViewItem const&, QModelIndex const&) const ()
   from /usr/lib/kvirc/4.1/modules/libkvilist.so
No symbol table info available.
#3  0x00007ffff61bab28 in QTreeView::indexRowSizeHint(QModelIndex const&) const
    () from /usr/lib/libQtGui.so.4
No symbol table info available.
#4  0x00007ffff61c3da7 in QTreeView::rowsInserted(QModelIndex const&, int, int)
    () from /usr/lib/libQtGui.so.4
No symbol table info available.
#5  0x00007ffff61804ad in QAbstractItemView::qt_metacall(QMetaObject::Call, int, void**) () from /usr/lib/libQtGui.so.4
No symbol table info available.
#6  0x00007ffff61c9775 in QTreeView::qt_metacall(QMetaObject::Call, int, void**) () from /usr/lib/libQtGui.so.4
No symbol table info available.
#7  0x00007ffff6200de5 in QTreeWidget::qt_metacall(QMetaObject::Call, int, void*---Type <return> to continue, or q <return> to quit---
*) () from /usr/lib/libQtGui.so.4
No symbol table info available.
#8  0x00000000006696e0 in KviThemedTreeWidget::qt_metacall(QMetaObject::Call, int, void**) ()
No symbol table info available.
#9  0x00007ffff4f66e3f in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () from /usr/lib/libQtCore.so.4
No symbol table info available.
#10 0x00007ffff4fb2314 in QAbstractItemModel::rowsInserted(QModelIndex const&, int, int) () from /usr/lib/libQtCore.so.4
No symbol table info available.
#11 0x00007ffff4f4d087 in QAbstractItemModel::endInsertRows() ()
   from /usr/lib/libQtCore.so.4
No symbol table info available.
#12 0x00007ffff61fb44c in QTreeWidgetItem::insertChild(int, QTreeWidgetItem*)
    () from /usr/lib/libQtGui.so.4
No symbol table info available.
#13 0x00007ffff61fb5fb in QTreeWidgetItem::addChild(QTreeWidgetItem*) ()
   from /usr/lib/libQtGui.so.4
No symbol table info available.
#14 0x00007ffff61fbf1d in QTreeWidgetItem::QTreeWidgetItem(QTreeWidget*, int)
    () from /usr/lib/libQtGui.so.4
No symbol table info available.
---Type <return> to continue, or q <return> to quit---
#15 0x00007fffdd4e8597 in ChannelTreeWidgetItem::ChannelTreeWidgetItem(QTreeWidget*, ChannelTreeWidgetItemData*) ()
   from /usr/lib/kvirc/4.1/modules/libkvilist.so
No symbol table info available.
#16 0x00007fffdd4e8b31 in ListWindow::flush() ()
   from /usr/lib/kvirc/4.1/modules/libkvilist.so
No symbol table info available.
#17 0x00007fffdd4ec0b4 in ListWindow::qt_metacall(QMetaObject::Call, int, void**) () from /usr/lib/kvirc/4.1/modules/libkvilist.so
No symbol table info available.
#18 0x00007ffff4f66e3f in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () from /usr/lib/libQtCore.so.4
No symbol table info available.
#19 0x00007ffff4f63a63 in QObject::event(QEvent*) ()
   from /usr/lib/libQtCore.so.4
No symbol table info available.
#20 0x00007ffff5c5922c in QApplicationPrivate::notify_helper(QObject*, QEvent*)
    () from /usr/lib/libQtGui.so.4
No symbol table info available.
#21 0x00007ffff5c5f6fb in QApplication::notify(QObject*, QEvent*) ()
   from /usr/lib/libQtGui.so.4
No symbol table info available.
#22 0x00007ffff7264a16 in KApplication::notify(QObject*, QEvent*) ()
---Type <return> to continue, or q <return> to quit---
   from /usr/lib/libkdeui.so.5
No symbol table info available.
#23 0x00007ffff4f5406c in QCoreApplication::notifyInternal(QObject*, QEvent*)
    () from /usr/lib/libQtCore.so.4
No symbol table info available.
#24 0x00007ffff4f80d42 in ?? () from /usr/lib/libQtCore.so.4
No symbol table info available.
#25 0x00007ffff4f7d848 in ?? () from /usr/lib/libQtCore.so.4
No symbol table info available.
#26 0x00007ffff0f518c2 in g_main_context_dispatch () from /lib/libglib-2.0.so.0
No symbol table info available.
#27 0x00007ffff0f55748 in ?? () from /lib/libglib-2.0.so.0
No symbol table info available.
#28 0x00007ffff0f558fc in g_main_context_iteration ()
   from /lib/libglib-2.0.so.0
No symbol table info available.
#29 0x00007ffff4f7d513 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQtCore.so.4
No symbol table info available.
#30 0x00007ffff5d0946e in ?? () from /usr/lib/libQtGui.so.4
No symbol table info available.
#31 0x00007ffff4f52992 in QEventLoop::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQtCore.so.4
---Type <return> to continue, or q <return> to quit---
No symbol table info available.
#32 0x00007ffff4f52d6c in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQtCore.so.4
No symbol table info available.
#33 0x00007ffff4f56aab in QCoreApplication::exec() ()
   from /usr/lib/libQtCore.so.4
No symbol table info available.
#34 0x0000000000643253 in main ()
No symbol table info available.
```

---

Resolution: fixed → none
Status: closed → reopened

**Comment by DarthGandalf** [comment]
**Comment by pragmaware on 10 Feb 2011 00:00:42**
fixed in #5446

---

Resolution: none → fixed
Status: reopened → closed
